lib/bolt/pal.rb

```diff
@@ -632,7 +633,19 @@ def run_plan(plan_name, params, executor, inventory = nil, pdb_client = nil, app
       Bolt::PlanResult.new(e, 'failure')
     end
 
-    def lookup(key, targets, inventory, executor, _concurrency)
+    def plan_hierarchy_lookup(key, vars = {})
+      # Do a lookup with a script compiler, which uses the 'plan_hierarchy' key in
+      # Hiera config.
+      with_puppet_settings do
+        # We want all of the setup and teardown that `in_bolt_compiler` does,
+        # but also want to pass keys to the script compiler.
+        in_bolt_compiler(compiler_params: { variables: vars }) do |compiler|
+          compiler.call_function('lookup', key)
+        end
+      end
+    end
+
+    def lookup(key, targets, inventory, executor, plan_vars: {})
```

Comment:
Small nitpick, but it seems weird to have `vars` be a positional arg for `plan_hierarchy_lookup`, but a keyword arg here.

Reply:
Ah yeah and to have different names. Updated, good catch!